sklearn/preprocessing/data.py

```diff
@@ -3147,3 +3150,38 @@ def inverse_transform(self, X):
                     X_tr[mask, idx] = None
 
         return X_tr
+
+    def get_feature_names(self, input_features=None):
+        """Return feature names for output features
+
+        Parameters
+        ----------
+        input_features : list of string, length n_features, optional
+            String names for input features if available. By default,
+            "x0", "x1", ... "xn_features" is used.
+
+        Returns
+        -------
+        output_feature_names : list of string, length n_output_features
+
+        """
+        is_python3 = sys.version_info.major == 3
+        if is_python3:
+            unicode = str
+
+        cats = self.categories_
+        feature_names = []
+        if input_features is None:
+            input_features = ['x%d' % i for i in range(len(cats))]
+        elif(len(input_features) != len(self.categories_)):
+            raise ValueError("input_features should have"
+                             " length equal to number of features")
+
+        def to_unicode(text):
+            return text if isinstance(text, unicode) else text.encode('utf8')
```

Comment:
I think you can use `six.text_type` here instead of the `unicode` (and redefining unicode depending on the version)